art/step/step.py

```diff
@@ -192,23 +194,23 @@ class ModelStep(Step):
 
     def __init__(
         self,
-        model: ArtModule,
+        model_func: Callable[[], ArtModule],
         trainer_kwargs: Dict = {},
         logger: Optional[Union[Logger, Iterable[Logger], bool]] = None,
     ):
         """
         Initialize a model-based step.
 
         Args:
-            model (ArtModule): The model associated with this step.
+            model_func (ArtModule): The model associated with this step.
             trainer_kwargs (Dict, optional): Arguments to be passed to the trainer. Defaults to {}.
             logger (Optional[Union[Logger, Iterable[Logger], bool]], optional): Logger to be used. Defaults to None.
         """
         super().__init__()
         if logger is not None:
             logger.add_tags(self.name)
-
-        self.model = model
+        assert isinstance(model_func, Callable)
```

Comment:
This is too week, basically any torch module is Callable

Reply:
![image](https://github.com/SebChw/art/assets/66476124/96610f56-9de1-42a6-8af2-abdc75d0dfca)